modules/apps/frontend-token/frontend-token-definition-impl/src/main/java/com/liferay/frontend/token/definition/internal/jaxrs/application/FrontendTokenDefinitionApplication.java

```diff
@@ -0,0 +1,214 @@
+import com.liferay.frontend.token.definition.internal.FrontendTokenDefinitionImpl;
+import com.liferay.frontend.token.definition.internal.validator.FrontendTokenDefinitionValidator;
+import com.liferay.portal.json.validator.JSONValidatorException;
+import com.liferay.portal.kernel.feature.flag.FeatureFlagManagerUtil;
+import com.liferay.portal.kernel.json.JSONException;
+import com.liferay.portal.kernel.json.JSONFactory;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.upload.UploadServletRequest;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.io.File;
+import java.io.IOException;
+
+import java.nio.file.Files;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Locale;
+import java.util.Set;
+
+import javax.servlet.http.HttpServletRequest;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.FormParam;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Application;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.jaxrs.whiteboard.JaxrsWhiteboardConstants;
+
+/**
+ * @author Anderson Luiz
+ * @author Thiago Buarque
+ */
+@Component(
+	property = {
+		JaxrsWhiteboardConstants.JAX_RS_APPLICATION_BASE + "=/com-liferay-frontend-token-definition-impl",
+		JaxrsWhiteboardConstants.JAX_RS_NAME + "=com.liferay.frontend.token.definition.internal.jaxrs.application.FrontendTokenDefinitionApplication",
+		"auth.verifier.auth.verifier.PortalSessionAuthVerifier.urls.includes=/*",
+		"auth.verifier.guest.allowed=false", "liferay.oauth2=false"
+	},
+	service = Application.class
+)
+public class FrontendTokenDefinitionApplication extends Application {
+
+	@Path("/tokens-info")
+	@POST
+	public Response getFrontendTokenDefinitionTokensInfo(
+		@Context HttpServletRequest httpServletRequest,
+		@FormParam("companyId") long companyId,
+		@FormParam("externalReferenceCode") String externalReferenceCode) {
+
+		FrontendTokenDefinition frontendTokenDefinition =
+			_frontendTokenDefinitionRegistry.getFrontendTokenDefinition(
+				companyId, externalReferenceCode, null);
+
+		if (frontendTokenDefinition != null) {
+			return _getFrontendTokenDefinitionTokensInfo(
+				_portal.getLocale(httpServletRequest), frontendTokenDefinition);
+		}
+
+		return Response.ok(
+		).build();
+	}
+
+	public Set<Object> getSingletons() {
+		return Collections.singleton(this);
+	}
+
+	@Consumes(MediaType.MULTIPART_FORM_DATA)
+	@Path("/validate-file")
+	@POST
+	@Produces(MediaType.APPLICATION_JSON)
+	public Response validateFrontendTokenDefinitionFile(
```

Comment:
Why don't we have this process the string value rather than the file. When this api is called we already have the string value, so it seams this would be more efficient.

Reply:
There is a size limit we encountered when sending the json representation of the file. From a quick search we found out the server can limite the body size but we did not find where it's defined (it seems to be a global config, not endpoint ranged).

The file we tried sending had 300kb. Sending smaller files worked fine.